### PR TITLE
only set cache-control headers for success response

### PR DESCRIPTION
**Pull request by fraserxu**
This pull request trying to fix two thing:
1. Remove default cache-control header https://github.com/divshot/cache-control/pull/1
2. Only set cache-control headers when the response statusCode is `2xx` as we don't want to cache error response in most case.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -10,76 +10,61 @@ var caches = {
...
   it('sets cache control to the passed string if specified in config file', function (done) {
-    
+
     request(app)
       .get('/private.html')
       .expect('Cache-Control', 'private, max-age=300')
       .end(done);
   });
-  
-  it('sets cache control to 5 minutes? by default', function(done) {
```
Comment (scottcorgan) [inline test/index.js:46]: Why did you delete these tests?

Comment (fraserxu) [reply test/index.js:46]: Hmm, so if we don't set default cache-control header, what we should check here in the test? 🤔

Comment (scottcorgan) [reply test/index.js:46]: There's a default set to 5 minutes

Comment (fraserxu) [reply test/index.js:46]: @scottcorgan It's intended to remove the default cache-headers for the same reason as #1 . The current code will override whatever is set inline which might not work as expected.

**Comment by scottcorgan** [comment]
Thanks. Published as `2.0.0`.

**Comment by fraserxu** [comment]
Thank you! 🎉
